core/database/foxx/tests/user_router.test.js

```diff
@@ -0,0 +1,130 @@
+"use strict"
+// NOTE: completion of tests requires successful run of user_fixture.js script
+
+// Integration test of API
+const { expect } = require("chai");
+const request = require("@arangodb/request");
+const { baseUrl } = module.context;
+const { db } = require("@arangodb");
+
+const usr_base_url = `${baseUrl}/usr`
+
+
+// NOTE: describe block strings are compared against test specification during test call, not file name
+describe("user_router: the Foxx microservice user_router token/set endpoint", () => {
+    const test_param_indexes = [0,1,2,3,4];
+    const test_params = test_param_indexes.map((param_index) => {
+        return {
+            _key: "testUser" + param_index,
+            access: "access_token" + param_index,
+            refresh: "refresh_token" + param_index,
+        }
+    });
+    const test_edge_params = {
+        type: 4,  // transfer token
+        other_token_data: "1cbaaee5-b938-4a4e-87a8-f1ec4d5d92f9",   // fake UUID
+    };
+
+    it("should accept a valid user's token and execute an update", () => {
+        // arrange
+        // NOTE: the get request has query params instead of a body
+        const query_params = test_params[0];
+        // TODO: make encoded query params less hard coded
+        const request_string = `${usr_base_url}/token/set?client=${query_params._key}&access=${query_params.access}&refresh=${query_params.refresh}&expires_in=500000`;
+
+        // act
+        const response = request.get(request_string);
+
+        // assert
+        expect(response.status).to.equal(204);
+
+        const user_token_data = db.u.document({_key: query_params._key});
+        expect(user_token_data).to.include(query_params);
+    });
+
+    it("should update only a valid user's token when additionally provided type", () => {
+        // arrange
+        const local_test_params = test_params[1];
+        const query_params = {
+          ...local_test_params,
+          type: 4,
+        };
+        // TODO: make encoded query params less hard coded
+        const request_string = `${usr_base_url}/token/set?client=${query_params._key}&access=${query_params.access}&refresh=${query_params.refresh}&expires_in=500000&type=${query_params.type}`;
+
+        // act
+        const response = request.get(request_string);
+
+        // assert
+        expect(response.status).to.equal(204);
+
+        const user_token_data = db.u.document({_key: query_params._key});
+        expect(user_token_data).to.include(local_test_params);
+
+        const user_token_edge = db.globus_token.outEdges(user_token_data._id);
+        expect(user_token_edge).to.be.empty;
+    });
+
+    it("should update only a valid user's token when additionally provided other token data", () => {
+        // arrange
```

Comment:
This test should also throw an error because including additional information that is not actually being stored or used by the route indicates the user doesn't understand what the route is supposed to be doing. In this case other info is being used at all.